TestEnvironment/demo-static-parent-issue/MyClass.php

```diff
@@ -10,6 +10,14 @@ public function load()
 		$refreshedEntity = self::refreshEntity($entity);
 		$refreshedEntity->entityMethod(); // auto-complete does not work for this
 	}
+
+	public function loadStaticProperty()
+	{
+		$user = StaticPropertyDataProvider::$user->getAccessToken();
+
+		$refreshedAccessToken = self::refreshEntity($user);
+		$refreshedAccessToken->isValid(); // auto-complete does not work for this
```

Comment:
try now...

In theory all method calls are now handled.



On 28 April 2016 at 15:02, Martin Hujer <notifications@github.com> wrote:

> In TestEnvironment/demo-static-parent-issue/MyClass.php
> <https://github.com/pbyrne84/DynamicReturnTypePluginTestEnvironment/pull/4#discussion_r61431714>
> :
>
> > @@ -10,6 +10,14 @@ public function load()
> >  		$refreshedEntity = self::refreshEntity($entity);
> >  		$refreshedEntity->entityMethod(); // auto-complete does not work for this
> >  	}
> > +
> > +	public function loadStaticProperty()
> > +	{
> > +		$user = StaticPropertyDataProvider::$user->getAccessToken();
> > +
> > +		$refreshedAccessToken = self::refreshEntity($user);
> > +		$refreshedAccessToken->isValid(); // auto-complete does not work for this
>
> @pbyrne84 <https://github.com/pbyrne84> JavaScript gets
> \StaticPropertyDataProvider.user.getAccessToken in this case
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/pbyrne84/DynamicReturnTypePluginTestEnvironment/pull/4/files/0997aa0a11327b9371750d012af02948211d8321#r61431714>
>